app/src/main/java/com/epfl/drawyourpath/database/FireDatabase.kt

```diff
@@ -380,54 +431,89 @@ class FireDatabase : Database() {
      * @param data the data snapshot containing the user List
      * @return a list containing the userIds of the friends
      */
-    private fun transformFriendsList(data: DataSnapshot?): List<String>{
-        if(data == null){
+    private fun transformFriendsList(data: DataSnapshot?): List<String> {
+        if (data == null) {
             return emptyList()
         }
         val friendListUserIds = ArrayList<String>()
 
-        for(friend in data.children){
+        for (friend in data.children) {
             friendListUserIds.add(friend.key as String)
         }
         return friendListUserIds
     }
 
+    /**
+     * Helper function to obtain the runs history from the database of the user
+     * @param data the data snapshot containing the history
+     * @return a list containing the history of the runs of the user
+     */
+    private fun transformRunsHistory(data: DataSnapshot?): List<Run> {
+        if (data == null) {
+            return emptyList()
+        }
+        val runsHistory = ArrayList<Run>()
+
+        for (run in data.children) {
+            val points = ArrayList<LatLng>()
+            for (point in run.child("path").child("points").children) {
+                val lat = point.child("latitude").value as Long
+                val lon = point.child("longitude").value as Long
```

Comment:
Don't we lose floating point data due to the conversion `Long` -> `Double`? Shouldn't it be read as `Double` directly?
Also, if there is no value in the database, it will crash as it calls `toDouble()` on a `null` value. We might want to handle this case.

Reply:
You are absolutely right, they should be read as Double, good catch. 
With this change however we have no call to toDouble() anymore. Do you still think I should check for null ? If I do so I should do it for every value I read, even though it should be impossible to store runs with points without latitude or longitude... What do you think ?

Reply:
I think we should still check for `null` in case the data is badly set in the database. Crashing the app with a `null pointer exception` might quickly become hard to trackdown for this case.

On another note, do you know how the type is handled by Firebase for a JSON `Number`? I've seen your test data is integers so I'm wondering if it will not return a `Long` in this case and a `Double` when it has decimals? If that's the case, we might need to handle both...

Reply:
Alright I will check for null points then, however if the start or end time is not correct I guess it will crash anyway. 
As I said I don't see how points could be badly formatted as they are basic LatLong objects when stored, so this should not be an issue and such runs should never exist.

I don't know how it is stored precisely, but will check with real values that are not integers and let you know. Taking my tests into consideration is not relevant though, as they use the mock database.

Reply:
@clementcharmillot  Sorry actually you are right I had to do a small change for it to work correctly with both types. Again good catch thank you!